### DISCUSSION TITLE
MB-Lab 1.8.0 info

### DISCUSSION THREAD

**Discussion by animate1978** [animate1978/MB-Lab#397] [General]
MB-Lab 1.8.0 is developmental for Blender 4.0 currently on my system. Had to do a little work to get it functioning again but at the basic level, works. I am not so sure about the MB-Dev side of things but assuming it to is at a functional state but I haven't tested it. The basic functions of MB-Lab do in fact run on my copy of Blender 4.0 otherwise.

On that note I want to stress that development will be slow, but MB-Lab has booted up again.

I think a lot of the issues people had with MB-Lab were fixed in the 'dev' branch that was never merged into the main, so people were trying for years to get MB-Lab to run but it was buggy, while I on the other hand tested my version and this seemed to run fine on Blender 3.0+.... either way, things at the end got messy and unfinished.

Some people have migrated away, which is fine. MB-Lab was always supposed to be a community and people come and go, this is normal. Especially when the project looks dark. So do I blame anyone for leaving MB-Lab? Not really.

MB-Lab 1.8.0 is not much different than previous versions other than the fact that it actually run on 4.0 but it will also REQUIRE Blender 4.0 mainly due to the PrincipledBSDF shader, as well as some code changes.

The skin shader will be a rework of the older 1.7.6 skin shader, simply because I don't have the time or energy to build one from scratch again. I have been working on it trying to achieve a better look. Mainly skin is a difficult thing to shade, it suffers from that uncanny valley too where we can tell if it looks fake, almost real or photo real. Unfortunately the skin shader has been one of the most frequently altered thing in this project but I THINK that the old 1.7.6 shader would be far more effective than using one based on the PrincipledBSDF shader. The current build of the shader has two specular lobes if you will, one for the skin shininess as well as surface oil specularity. It has two different SSS shaders, one for two different layers of skin such as the top layer, and the more fatty tissue below. The problem is the procedural stuff I had made doesn't want to work well, so I have essentially shut some of them off temporarily.

The Hair presets? Those are buggy and unfortunately don't know how to fix them, that was a project by another dev. Just warning that right now they will not function as intended due to again changes.... The hair shows up but it renders incorrectly.

The Auto-Updater will be removed for the time being.

I don't know, just trying to keep it alive, it seems to still get attention even after going silent for so long.

**Comment by 1nky** [comment]
Hey man I think it's great you still going for it ...It;s a worthwhile project and something blender needs.
spent the day looking for something half way decent to make some characters for a project i want to do and came back to the plugin (running daz in bottles for Linux was almost as jankey as daz is to use )
I'm using blender 4 and just about got MBl to work ,,, if you like i could help with bug reporting etc ...  I can build a model its usable and the rigging works well, hair not so good  

I have thought about a add-on that  would generate characters with geometry nodes perhaps you could use geonodes to deal with some of the "procedural stuff " 
all the best

**Reply by animate1978** [reply]
The particle hair is buggy to say the least, it kind of always has been but Noizirom did his best to try to make a workable system. The main issue with the Hair is that it is using older PrincipledHairBSDF node inside the .npz file, which I believe is just a compressed file I just have no clue how to work with them.

A couple of years ago when the GeoNode system was first released some of us thought about using it to build characters as well, however the problem is that it would be just incredibly difficult to implement, just with the variations of human anatomy + keeping everything contained in a nice form without seams, at that time it just didn't work well. Organic shapes I think would be difficult to construct from pieces. Not to mention I have limited experience with GeoNodes so I would totally NOT be a dev in that aspect lol.

There are still some bugs that persist from previous versions, minor and not breaking they just don't work well.
https://github.com/animate1978/MB-Lab/issues/360
https://github.com/animate1978/MB-Lab/issues/317
https://github.com/animate1978/MB-Lab/issues/358

Related to the rest poses and proxy fitting... unfortunately locating these bugs in the code has been difficult.

**Comment by Reasonant** [comment]
Hi there guys. I am a huge fan of this add on and generally of Blender. I am studying Artificial Intelligence on IU international University of Applied Sciences (Germany). (I am doing it to find a better job.) Anyway. I would like to learn to code by doing something that interests me. I am trying to learn alone but ... it is difficult. So, I would like to ask if I could somehow help with this project. I would need someone to explain to me some basic stuff and the steps I could follow to contribute to this project. 

I also have some very vauge ideas of implementing machine learning techniques to produce more accurate results. I do not yet know exactly how we could achieve this or what should we achieve anyway. Though I imagine a completely professional character creation add-on, which can produce idnustry level characters. Anyway I just wanted to share this last ideas.

It would be fun to code for this project, so I can help tell me.

**Reply by animate1978** [reply]
Of course this addon could always use more help and since the project went into hibernation for some time, pretty much everyone who had been involved no longer is so yes please!!

Right now I am gearing up for releasing the next version around the time that Blender 4.1 is released, so there is not much to do other than make sure things work as intended. Currently only a couple of things remain buggy I think. At the moment I have been working on the surface shaders myself so that's been my priority lately. The code that I changed to get it to run in Blender 4.0 is pretty much hacks or commented out stuff.

For machine learning stuff, I am not too sure what COULD be implemented in terms of what it could do for the addon. It's a tough gray area.

So in order to really get involved I would suggest downloading Blender 4.0 or 4.1 Beta, and the 4_0_dev branch of this addon here : 

https://github.com/animate1978/MB-Lab/tree/4_0_dev_branch

There is some basic documentation, that really needs updating badly, here : 

https://mb-lab-docs.readthedocs.io/en/latest/index.html

Hopefully this helps get you started. If you have any questions feel to ask.

**Comment by animate1978** [comment]
So MB-Lab 1.8.0 was quietly released yesterday and hastily of course naturally without testing everything possible under the sun, the ONE thing I added that was supposed to be a new thing was rendered incorrectly and now I have to work out the shader math and figure out what I did wrong.

So this means that 1.8.1 is right around the corner! lol.

Anyways I did do a lot of testing which is why the last 2 months of development was slow but I rushed the last bit because I felt that I was taking too long in releasing this and now I got another botched release, because I have done this before and I was so sure of this one I forgot to check the skin shader before pushing the last update.

Not only that but I also botched it by releasing with a dot tag, such MB-Lab 1.8.0 which has caused many problems before! In the releases since I have used 1_7_9 for instance. This way when people download it, it installs correctly in Blender. Otherwise you get an error which is exactly what happened to one such user that did open up an issue, did figure it out and then closed the issue but still ... how many other people downloaded the zip and got upset or gave up? This is why a few years ago I had bought a domain and rented a webhost for MB-Lab it just got so cumbersome to upkeep and secure... I eventually closed it. Plus I wasn't really working on MB-Lab seriously as well.

Now I have kind of balanced a workflow between my other 3D work, MB-Lab and it's support.... I have devoted at least a couple of hours a week on the project, I feel that I am ok to continue work on this now that it's been updated for 4.0.

This brings to mind the OTHER half of MB-Lab that I think NOBODY has touched... the MB-Dev tools. I will be uploading my work I did to a CC0 model I had found and spent I think like 2 years on trying to incorporate her into MB-Lab. I was successful to a degree until it came to the sculpting of the morphs. Then the project came to a halt. I couldn't do more to it. I feel ashamed saying this but I suck at sculpting. So until the morphs get some love that project only exists as an example of what MB-Dev can do.
But I also was thinking of recording some sort of series of videos of HOW to work with the MB-Dev tools... I think that is another reason I haven't seen anything come from it...nobody knows how to actually do it, except for the developer that created the tools and myself.

Anyways long post over. MB-Lab 1.8.0 is available for download so check it out and let me know if more bugs pop up.
